node/src/ledger/read/ledger_read_reducer.rs

```diff
@@ -1,20 +1,419 @@
-use super::{LedgerReadAction, LedgerReadActionWithMetaRef, LedgerReadState};
+use mina_p2p_messages::v2;
+use openmina_core::requests::RequestId;
+use p2p::{
+    channels::{
+        rpc::{P2pChannelsRpcAction, P2pRpcId, P2pRpcRequest, P2pRpcResponse},
+        streaming_rpc::{P2pChannelsStreamingRpcAction, P2pStreamingRpcRequest},
+    },
+    P2pAction, PeerId,
+};
+use redux::Dispatcher;
+
+use crate::{
+    block_producer::vrf_evaluator::BlockProducerVrfEvaluatorAction,
+    ledger_effectful::LedgerEffectfulAction, Action, RpcAction, State, Substate,
+};
+
+use super::{
+    LedgerAddress, LedgerReadAction, LedgerReadActionWithMetaRef, LedgerReadIdType,
+    LedgerReadInitCallback, LedgerReadRequest, LedgerReadResponse,
+    LedgerReadStagedLedgerAuxAndPendingCoinbases, LedgerReadState,
+};
 
 impl LedgerReadState {
-    pub fn reducer(&mut self, action: LedgerReadActionWithMetaRef<'_>) {
+    pub fn reducer(mut state_context: Substate<Self>, action: LedgerReadActionWithMetaRef<'_>) {
         let (action, meta) = action.split();
+        let Ok(ledger_read_state) = state_context.get_substate_mut() else {
```

Comment:
we (almost?) always call it just `state`, is there any good reason for this one to be the exception?